src/util/AcceptParser.ts

```diff
@@ -0,0 +1,219 @@
+  const result = input.replace(/"(?:[^"\\]|\\.)*"/gu, (match): string => {
+    const replacement = `"${idx}"`;
+    replacements[replacement] = match;
+    idx += 1;
+    return replacement;
+  });
+  return { result, replacements };
+};
+
+const splitAndClean = (input: string): string[] =>
+  input.split(',')
+    .map((part): string => part.trim())
+    .filter((part): boolean => part.length > 0);
+
+/**
+ * Parses a single media range with corresponding parameters from an Accept header.
+ * For every parameter value that is a double quoted string,
+ * we check if it is a key in the replacements map.
+ * If yes the value from the map gets inserted instead.
+ * @param part - A string corresponding to a media range and its corresponding parameters.
+ * @param replacements - The double quoted strings that need to be replaced.
+ *
+ * @returns {@link Accept} object corresponding to the header string.
+ */
+const parseAcceptPart = (part: string, replacements: { [id: string]: string }): Accept => {
+  const [ range, ...parameters ] = part.split(';').map((param): string => param.trim());
+  let weight = 1;
+  const mediaTypeParams: { [key: string]: string } = {};
+  const extensionParams: { [key: string]: string } = {};
+  let map = mediaTypeParams;
+  parameters.forEach((param): void => {
+    const [ name, value ] = param.split('=');
+    let actualValue = value;
+
+    if (value && value.length > 0 && value.startsWith('"') && replacements[value]) {
+      actualValue = replacements[value];
+    }
+
+    if (name === 'q') {
+      // Extension parameters appear after the q value
+      map = extensionParams;
+      weight = parseFloat(actualValue);
+    } else {
+      // Value is optional for extension parameters
+      map[name] = actualValue || '';
+    }
+  });
+
+  return {
+    range,
+    weight,
+    parameters: {
+      mediaType: mediaTypeParams,
+      extension: extensionParams,
+    },
+  };
+};
+
+/**
+ * Parses an Accept-* header where each part is only a value and a weight, so roughly /.*(q=.*)?/ separated by commas.
+ * @param input - Input header string.
+ *
+ * @returns An array of ranges and weights.
+ */
+const parseNoParameters = (input: string): { range: string; weight: number }[] => {
+  const parts = splitAndClean(input);
+
+  return parts.map((part): { range: string; weight: number } => {
+    const [ range, qvalue ] = part.split(';').map((param): string => param.trim());
+    const result = { range, weight: 1 };
+    if (qvalue) {
+      result.weight = parseFloat(qvalue.split('=')[1]);
+    }
+    return result;
+  }).sort((left, right): number => right.weight - left.weight);
+};
+
+// EXPORTED FUNCTIONS
+
+/**
+ * Parses an Accept header string.
+ * No validation is done so this assumes a valid input string.
+ *
+ * @param input - The Accept header string.
+ *
+ * @returns An array of {@link Accept} objects, sorted by weight.
+ */
+export const parseAccept = (input: string): Accept[] => {
```

Comment:
Because by default the linter is set up to only allow arrow functions due to the scoping differences between the two, and I couldn't come up with an argument why this would be better :D